### PR TITLE
Run kubernetes tests in parallel

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: scripts/ci/kubernetes/ci_setup_clusters_and_run_kubernetes_tests_in_parallel.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    parallel::kill_monitor
+    set -e
+    start_end::group_end
+}
+
+initialization::set_output_color_variables
+
+parallel::make_sure_gnu_parallel_is_installed
+parallel::make_sure_python_versions_are_specified
+parallell::make_sure_kubernetes_versions_are_specified
```
Comment (mik-laj) [inline scripts/ci/kubernetes/ci_setup_clusters_and_run_kubernetes_tests_in_parallel.sh]:
```suggestion
parallel::make_sure_kubernetes_versions_are_specified
```

**Comment by potiuk** [comment]
yet another improvement in paralell tests . And it nicely works in Public Runners as well: https://github.com/potiuk/airflow/runs/2274880512?check_suite_focus=true

I am going to close/reopen this one with 'full-test-needed' flag to see the improvements.

**Comment by potiuk** [comment]
Nice ! Parallel K8S run 3xdifferent combinations of K8S version + Python in 17 minutes. Previoyusly just 1 of the K8S runs run in 14 minutes. REady to merge!

**Comment by potiuk** [comment]
@mik-laj @kaxil

**Comment by potiuk** [comment]
Last Parallel improvement for a while I think (maybe I perallelize image builds but this is a bit more tricky)

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
Wooho!. Almost evyrything sped up

**Pull request merged**